contracts/examples/mintable-erc20-transfer/FxMintableERC20RootTunnel.sol

```diff
@@ -2,35 +2,70 @@
 pragma solidity ^0.8.0;
 
 import { Create2 } from "../../lib/Create2.sol";
-import { SafeMath } from "../../lib/SafeMath.sol";
-import { FxERC20 } from "../../tokens/FxERC20.sol";
+import { FxERC20Token, ERC20Token } from "../../tokens/FxERC20Token.sol";
 import { FxBaseRootTunnel } from "../../tunnel/FxBaseRootTunnel.sol";
-import {SafeERC20,IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { SafeERC20, IERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 
 /** 
  * @title FxMintableERC20RootTunnel
  */
 contract FxMintableERC20RootTunnel is FxBaseRootTunnel, Create2 {
-    using SafeMath for uint256;
     using SafeERC20 for IERC20;
 
     // maybe DEPOSIT and MAP_TOKEN can be reduced to bytes4
     bytes32 public constant DEPOSIT = keccak256("DEPOSIT");
-    //bytes32 public constant MAP_TOKEN = keccak256("MAP_TOKEN");
+    bytes32 public constant MAP_TOKEN = keccak256("MAP_TOKEN");
 
     mapping(address => address) public rootToChildTokens;
     address public rootTokenTemplate;
     bytes32 public childTokenTemplateCodeHash;
 
-     constructor(address _checkpointManager, address _fxRoot, address _rootTokenTemplate) FxBaseRootTunnel(_checkpointManager, _fxRoot) {
+     constructor(address _checkpointManager, address _fxRoot, address _rootTokenTemplate, address _childTokenTemplate) FxBaseRootTunnel(_checkpointManager, _fxRoot) {
         rootTokenTemplate = _rootTokenTemplate;
+        // compute child token template code hash
+        childTokenTemplateCodeHash = keccak256(
+            minimalProxyCreationCode(_childTokenTemplate)
+        );
     }
 
+    function mapToken(address rootToken) public {
```

Comment:
It is best to have a single set of bridge(Root tunnel & Child tunnel) contracts that handles all the tokens, so that we have a canonical mapping of the root token and child token, and a single point where bridged tokens are locked/minted.

Plus, the token contract being used ([`ERC20Token`](https://github.com/coinvise/fx-portal/pull/1/files#diff-86af3bc0904ec9e6825c2dc8ee0d33cff209938e6a2d65d43789040336ddf36dR56)) is different from the original token contract that we've [used](https://github.com/coinvise/fx-portal/blob/main/contracts/examples/erc20-transfer/FxERC20RootTunnel.sol#L61). Thus we're also using the [`creator`](https://github.com/coinvise/fx-portal/pull/1/files#diff-86af3bc0904ec9e6825c2dc8ee0d33cff209938e6a2d65d43789040336ddf36dR57) field in the token contract and [passing](https://github.com/coinvise/fx-portal/pull/1/files#diff-86af3bc0904ec9e6825c2dc8ee0d33cff209938e6a2d65d43789040336ddf36dR64) it along with the message to the child tunnel contract